app/helpers/physical_server_helper/textual_summary.rb

```diff
@@ -91,15 +91,24 @@ def textual_power_state
   end
 
   def textual_mac
-    {:label =>  _("Mac Address"), :value => @record.hardware.guest_devices.collect { |device| device[:address] }.join(", ") }
+    # Currently, we only want to display the MAC addresses of devices that
+    # are associated with an entry in the networks table. This ensures that
+    # we only display management addresses.
+    {:label =>  _("MAC Address"), :value => @record.hardware.guest_devices.reject { |device| device.network.nil? }.collect { |device| device[:address] }.join(", ") }
   end
 
   def textual_ipv4
-    {:label =>  _("IPV4 Address"), :value => @record.hardware.guest_devices.collect { |device| device.network.ipaddress }.join(", ") }
+    # It is possible for guest devices not to have network data (or a network
+    # hash). As a result, we need to exclude guest devices that don't have
+    # network data to prevent a nil class error from occurring.
+    {:label =>  _("IPv4 Address"), :value => @record.hardware.guest_devices.reject { |device| device.network.nil? }.collect { |device| device.network.ipaddress }.join(", ") }
```

Comment:
@AparnaKarve Thank you for the suggestion. I tried it, but it appears that because _try_ returns _nil_ when _device.network_ is _nil_, an empty string is being included in the final joined string. For example, I am seeing "10.243.6.17, " as output when one of the guest devices doesn't have network data.